src/ProducerSideService/include/ProducerSideService/BuildAndStartProducerSideServer.h

```diff
@@ -5,10 +5,34 @@
 #ifndef ORBIT_PRODUCER_SIDE_SERVICE_BUILD_AND_START_PRODUCER_SIDE_SERVER_H_
 #define ORBIT_PRODUCER_SIDE_SERVICE_BUILD_AND_START_PRODUCER_SIDE_SERVER_H_
 
+#include <memory>
+
+#include "OrbitBase/Result.h"
 #include "ProducerSideService/ProducerSideServer.h"
 
 namespace orbit_producer_side_service {
-[[nodiscard]] std::unique_ptr<ProducerSideServer> BuildAndStartProducerSideServer();
+
+// This is a temporary type that can implicitly convert to either the new return type or the old
+// one. It is required to keep the cloud collector build working until we can change the internal
+// code to the new type. The new return type will be
+// `ErrorMessageOr<std::unique_ptr<ProducerSideServer>>`.
+class BuildAndStartReturnValue : public ErrorMessageOr<std::unique_ptr<ProducerSideServer>> {
+ public:
+  template <typename... Args>
+  BuildAndStartReturnValue(Args&&... args)  // NOLINT
+      : ErrorMessageOr{std::forward<Args>(args)...} {}
+
+  // NOLINTNEXTLINE
```

Comment:
Just curious XD. Why this is needed here?

Reply:
Not needed. But silences the clang-tidy warning that warns us about implicit conversion operators.